packages/taquito/test/contract/contract-on-chain-view.spec.ts

```diff
@@ -9,10 +9,300 @@ import {
+describe('OnChainView test on K protocol', () => {
   let view: OnChainView;
   let mockRpcClient: {
+    getProtocols: jest.Mock<any, any>;
+    getChainId: jest.Mock<any, any>;
+    runScriptView: jest.Mock<any, any>;
+    getStorage: jest.Mock<any, any>;
+  };
+  let mockReadProvider: any;
+  beforeEach(() => {
+    mockRpcClient = {
+      getProtocols: jest.fn(),
+      getChainId: jest.fn(),
+      runScriptView: jest.fn(),
+      getStorage: jest.fn(),
+    };
+
+    mockRpcClient.getProtocols.mockResolvedValue({ protocol: Protocols.PtKathman });
+
+    mockRpcClient.getChainId.mockResolvedValue('test_chain_id');
+
+    mockRpcClient.runScriptView.mockResolvedValue({
+      data: { int: '23' },
+    });
+    mockRpcClient.getStorage.mockResolvedValue({ int: '3' });
+
+    const context = new Context(mockRpcClient as any);
+    mockReadProvider = new RpcReadAdapter(context);
+
+    view = new OnChainView(
+      mockRpcClient as any,
+      mockReadProvider,
+      'contractAddress',
+      new ViewSchema([
+        { string: 'add' },
+        { prim: 'nat' },
+        { prim: 'nat' },
+        [{ prim: 'UNPAIR' }, { prim: 'ADD' }],
+      ]),
+      { prim: 'nat' },
+      20
+    );
+  });
+
+  it('OnChainView is instantiable', () => {
+    expect(view).toBeInstanceOf(OnChainView);
+  });
+
+  it('should extract the signature of the view', () => {
+    expect(view.getSignature()).toEqual({
+      parameter: 'nat',
+      result: 'nat',
+    });
+  });
+
+  it('should execute of the view', async (done) => {
+    expect(
+      await view.executeView({
+        viewCaller: 'KT1TRHzT3HdLe3whe35q6rNxavGx8WVFHSpH',
+      })
+    ).toEqual(new BigNumber(23));
+
+    expect(mockRpcClient.runScriptView).toHaveBeenCalledWith({
+      contract: 'contractAddress',
+      view: 'add',
+      input: { int: '20' },
+      chain_id: 'test_chain_id',
+      source: 'KT1TRHzT3HdLe3whe35q6rNxavGx8WVFHSpH',
+    });
+
+    done();
+  });
+
+  it('should adapt the instructions SENDER, SELF_ADDRESS, BALANCE and AMOUNT to the context before executing the view', async (done) => {
+    const view = new OnChainView(
+      mockRpcClient as any,
+      mockReadProvider,
+      'contractAddress',
+      new ViewSchema([
+        { string: 'viewName' },
+        { prim: 'nat' },
+        { prim: 'nat' },
+        [{ prim: 'SENDER' }, { prim: 'SELF_ADDRESS' }, { prim: 'BALANCE' }, { prim: 'AMOUNT' }],
+      ]),
+      { prim: 'nat' },
+      20
+    );
+    await view.executeView({
+      viewCaller: 'KT1TRHzT3HdLe3whe35q6rNxavGx8WVFHSpH',
+    });
+    expect(mockRpcClient.runScriptView).toHaveBeenCalledWith({
+      contract: 'contractAddress',
+      view: 'viewName',
+      input: { int: '20' },
+      chain_id: 'test_chain_id',
+      source: 'KT1TRHzT3HdLe3whe35q6rNxavGx8WVFHSpH',
+    });
+
+    done();
+  });
+
+  it('should adapt the view instructions to the context before executing it when there are nested instructions to replace', async (done) => {
```

Comment:
This test is not relevant for K, because adaptViewCodeToContext is not called in this case